src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSinkConfig.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.bigtable.sink;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.etl.api.validation.InvalidConfigPropertyException;
+import io.cdap.plugin.gcp.common.GCPReferenceSourceConfig;
+
+import java.io.File;
+
+/**
+ * Holds configuration required for configuring {@link BigtableSink}.
+ */
+public final class BigtableSinkConfig extends GCPReferenceSourceConfig {
+  public static final String TABLE = "table";
+  public static final String INSTANCE = "instance";
+  public static final String KEY_ALIAS = "keyAlias";
+  public static final String COLUMN_FAMILY = "columnFamily";
+
+  @Name(TABLE)
+  @Macro
+  @Description("The table to read from. A table contains individual records organized in rows. "
+    + "Each record is composed of columns (also called fields). "
+    + "Every table is defined by a schema that describes the column names, data types, and other information.")
+  final String table;
+
+  @Name(INSTANCE)
+  @Macro
+  @Description("BigTable instance id. " +
+    "Uniquely identifies BigTable instance within your Google Cloud Platform project.")
+  final String instance;
+
+  @Name(KEY_ALIAS)
+  @Description("Name of the field for row key.")
+  @Macro
+  final String keyAlias;
+
+  @Name(COLUMN_FAMILY)
+  @Description("Column family to use for all inserted rows.")
+  @Macro
+  final String columnFamily;
+
+  public BigtableSinkConfig(String table,
+                            String instance,
+                            String keyAlias,
+                            String columnFamily) {
+    this.table = table;
+    this.instance = instance;
+    this.keyAlias = keyAlias;
+    this.columnFamily = columnFamily;
+  }
+
+  public void validate() {
+    super.validate();
+    if (!containsMacro(PROJECT) && null == project) {
```

Comment:
this won't actually try to detect the project. The getProject() method will try to detect and throw an exception if it can't read it. This should be:

```
if (!containsMacro(PROJECT)) {
  getProject();
}
```

I think I left a similar comment for the other PR, did we fix it there?

Reply:
yes, that was fixed in source. Missed this update when moved fixes to sink part. Thanks.